cpp/src/interop/detail/arrow_allocator.cpp

```diff
@@ -16,9 +16,40 @@
 
 #include <cudf/detail/interop.hpp>
 
+#include <memory>
+#include <sys/mman.h>
+#include <unistd.h>
+
 namespace cudf {
 namespace detail {
 
+/*
+  Enable Transparent Huge Pages (THP) for large (>4MB) allocations.
+  `buf` is returned untouched.
+  Enabling THP can improve performance of device-host memory transfers
+  significantly, see <https://github.com/rapidsai/cudf/pull/13914>.
+*/
+template <typename T>
+T enable_hugepage(T buf)
+{
+  if (buf->size() < (1u << 22u)) {  // Smaller than 4 MB
+    return buf;
+  }
+
+#ifdef MADV_HUGEPAGE
+  const auto pagesize = sysconf(_SC_PAGESIZE);
+  void* addr          = const_cast<uint8_t*>(buf->data());
+  if (addr == nullptr) { return buf; }
+  std::size_t length{static_cast<std::size_t>(buf->size())};
```

Comment:
The type is already spelled out in the same line
```suggestion
  auto length{static_cast<std::size_t>(buf->size())};
```